src/main/java/lotto/domain/WinningType.java

```diff
@@ -0,0 +1,49 @@
+package lotto.domain;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum WinningType {
+    FIRST(2_000_000_000, 6),
+    SECOND(30_000_000, 5),
+    THIRD(1_500_000, 5),
+    FOURTH(50_000, 4),
+    FIFTH(5_000, 3),
+    NOTHING(0, 0);
+
+    private final long prize;
+    private final int matchCount;
+
+    WinningType(long prize, int matchCount) {
+        this.prize = prize;
+        this.matchCount = matchCount;
+    }
+
+    public static WinningType findType(int matchCount, boolean matchBonus) {
+        if (matchCount == 5 && !matchBonus) {
+            return WinningType.THIRD;
+        }
+
+        return Stream.of(values())
+                .filter(it -> it.matchCount == matchCount)
+                .findFirst()
+                .orElse(NOTHING);
+    }
+
+    public static List<WinningType> getReverseValues() {
+        return Stream.of(values())
+                .sorted(Collections.reverseOrder())
+                .filter(type -> type != WinningType.NOTHING)
```

Comment:
다음과 같이 하면 소소한 성능 향상을 얻을 수 있을 것 같아요! 😄 
```suggestion
                .filter(type -> type != WinningType.NOTHING)
                .sorted(Collections.reverseOrder())                
```

Reply:
아하-!
매의 눈이십니다 +_+ 
감사합니다👍👍